### DISCUSSION TITLE
Upgraded tool

### DISCUSSION THREAD

**Discussion by EONRaider** [opsdisk/pagodo#49] [Ideas]
Hey. So I found this tool a few days ago and somehow had this rather ambitious idea of upgrading it into a full dorking tool capable of dealing not only with Google but also with Shodan and Github dorks (to start with).

The end result would still be a CLI tool but it would also be cool to have it as a containerized application ready to deploy on a VPS or cloud infrastructure. This can be left to the end, though, and we don't need to worry about it right now.

Having this end goal in mind I forked this project and started to heavily refactor it so we can have a core application that is modular and robust enough to accept these modifications and added features. I'll be able to push the initial draft to my fork in a few days.

Would you be willing to work on such a thing in collaboration?

**Comment by opsdisk** [comment]
Hey @EONRaider 

Appreciate you taking the time to submit a discussion and check out the tool!

In general, I believe in the Unix philosophy of "doing one thing and doing it well" and that applies to this tool.  I'm certainly willing to entertain a more modular and generalized framework that may work with other separate Shodan and Github dork tools, but I don't want to add feature bloat to this one.  It's a delicate balance between creating a tool that does all the things and one that wraps a couple of tools together nicely.

That being said, if you still want to submit a PR so I can better understand what you're talking about, that'd still be cool.  It's possible I don't see the vision and potential of what you're proposing, so I'm definitely open to being proven wrong.

A containerized app would be great and I'd accept a PR for that with the current code base!
